src/framework/script/script-type.js

```diff
@@ -234,6 +260,61 @@ class ScriptType extends EventHandler {
         return this._enabled && !this._destroyed && this.entity.script.enabled && this.entity.enabled;
     }
 
+    /**
+     * Listen for an event on an EventHandler. The ScriptType will add and remove the event listeners on the
+     * EventHandler automatically when the ScriptType is enabled, disabled and destroyed.
+     *
+     * @param {EventHandler} eventHandler - EventHandler to listen for events on.
+     * @param {string} name - Name of the event to bind the callback to.
+     * @param {HandleEventCallback} callback - Function that is called when event is fired. Note
+     * the callback is limited to 8 arguments.
+     * @param {object} [scope] - Object to use as 'this' when the event is fired.
+     * @example
+     * this.listen(this.app.mouse, 'mousemove', this.onMouseMove, this);
+     * @see {@link ScriptType#unlisten} to remove listeners to an event on an EventHandler.
+     */
+    listen(eventHandler, name, callback, scope) {
+        this._listeners.push({
+            eventHandler,
+            name,
+            callback,
+            scope
+        });
+
+        if (this.enabled) {
+            eventHandler.on(name, callback, scope);
+        }
+    }
+
+    /**
+     * Removes a listener for an event on an EventHandler that was added by {@link ScriptType#listen}.
+     *
+     * @param {EventHandler} eventHandler - EventHandler that was originally listened to.
+     * @param {string} name - Name of the event that was originally listened to.
+     * @param {HandleEventCallback} callback - Function that was used as the callback when the event was originally
+     * listened to.
+     * @param {object} [scope] - Object that was used as the scope when the event was originally listened to
+     * @example
+     * this.unlisten(this.app.mouse, 'mousemove', this.onMouseMove, this);
+     * @see {@link ScriptType#listen} to listen to events on an EventHandler.
+     */
+    unlisten(eventHandler, name, callback, scope) {
+        let found = false;
+        for (let i = 0; i < this._listeners.length; ++i) {
+            const l = this._listeners[i];
+            if (l.eventHandler === eventHandler && l.name === name && l.callback === callback && l.scope === scope) {
+                this._listeners.splice(i, 1);
+                eventHandler.off(name, callback, scope);
+                found = true;
+                break;
+            }
+        }
+
+        if (!found) {
+            Debug.warn(`${this.__name}: Cannot find listener to remove for event ${name}. Please check where you originally listened for the event.`);
+        }
+    }
```

Comment:
I also don't consider using this too often and performance is a one-time-thing and doesn't even look too bad... it could be a bit shorter tho?

```suggestion
    unlisten(eventHandler, name, callback, scope) {
        for (let i = 0; i < this._listeners.length; ++i) {
            const l = this._listeners[i];
            if (l.eventHandler === eventHandler && l.name === name && l.callback === callback && l.scope === scope) {
                this._listeners.splice(i, 1);
                eventHandler.off(name, callback, scope);
                return;
            }
        }
        Debug.warn(`${this.__name}: Cannot find listener to remove for event ${name}. Please check where you originally listened for the event.`);
    }
```

Reply:
Hmm, not sure why I used a flag for found 🤔

Reply:
Now I remember, I used to allow chaining so there was a `return this;` at the end of the function. I removed it as I rarely see chaining used in practise where it already exists and didn't want to limit future expansion of the API unnecessarily

Reply:
Good catch, I've taken on your suggestion